apps/tests/aws-runtime/test/test-service.ts

```diff
@@ -672,6 +675,109 @@ export const transactionWorkflow = workflow(
   }
 );
 
+export const myBucket = bucket("myBucket");
+export const bucketSignal = signal<{ data: string }>("bucketSignal");
+
+export const myBucketHandler = myBucket.on(
+  "put",
+  "myBucketHandler",
+  { filters: [{ prefix: "key/" }] },
+  async (item) => {
+    const executionId = item.key.slice(4);
+    const obj = await myBucket.get(item.key);
+
+    if (obj?.body) {
+      await bucketSignal.sendSignal(executionId, {
+        data: await streamToString(obj.body),
+      });
+    }
+  }
+);
+
+async function streamToString(stream: stream.Readable) {
+  // lets have a ReadableStream as a stream variable
+  const chunks = [];
+
+  for await (const chunk of stream) {
+    chunks.push(Buffer.from(chunk));
+  }
+
+  return Buffer.concat(chunks).toString("utf-8");
+}
```

Comment:
Is the returned object always a stream?

No way should we expect customers to always do this. The library should have two different methods, one for stream, one for string/Buffer?

Reply:
We could add a method to the returned stream that stringifies it, `toString("utf8")`, `toBuffer()`. Is there any precedent for this?

Reply:
V3 returns a stream.Readable in node.

The return type from V3 has a transformToString on it.

https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/index.html#streams

Reply:
```ts
/**
 * The interface contains mix-in utility functions to transfer the runtime-specific
 * stream implementation to specified format. Each stream can ONLY be transformed
 * once.
 */
export interface SdkStreamMixin {
    transformToByteArray: () => Promise<Uint8Array>;
    transformToString: (encoding?: string) => Promise<string>;
    transformToWebStream: () => ReadableStream;
}
```

Reply:
Lets do....

`get(key).toString(encoding?: string)` which calls `transformToString` and caches the first result.

Reply:
Added